CI:  mypy validation failing on master

Happening in multiple PRs, e.g., https://github.com/pandas-dev/pandas/pull/41203/checks?check_run_id=3181637542
```
pandas/core/reshape/tile.py:421: error: No overload variant of "searchsorted" of "ndarray" matches argument types "Any", "str"  [call-overload]
pandas/core/reshape/tile.py:421: note: Possible overload variants:
pandas/core/reshape/tile.py:421: note:     def searchsorted(self, v: Union[int, float, complex, str, bytes, generic], side: Union[Literal['left'], Literal['right']] = ..., sorter: Union[_SupportsArray[dtype[Union[bool_, integer[Any]]]], Sequence[_SupportsArray[dtype[Union[bool_, integer[Any]]]]], Sequence[Sequence[_SupportsArray[dtype[Union[bool_, integer[Any]]]]]], Sequence[Sequence[Sequence[_SupportsArray[dtype[Union[bool_, integer[Any]]]]]]], Sequence[Sequence[Sequence[Sequence[_SupportsArray[dtype[Union[bool_, integer[Any]]]]]]]], int, Sequence[int], Sequence[Sequence[int]], Sequence[Sequence[Sequence[int]]], Sequence[Sequence[Sequence[Sequence[int]]]], None] = ...) -> signedinteger[Any]
pandas/core/reshape/tile.py:421: note:     def searchsorted(self, v: Union[Sequence[Sequence[Sequence[Sequence[Sequence[Any]]]]], Union[Union[_SupportsArray[dtype[Any]], Sequence[_SupportsArray[dtype[Any]]], Sequence[Sequence[_SupportsArray[dtype[Any]]]], Sequence[Sequence[Sequence[_SupportsArray[dtype[Any]]]]], Sequence[Sequence[Sequence[Sequence[_SupportsArray[dtype[Any]]]]]]], Union[bool, int, float, complex, str, bytes, Sequence[Union[bool, int, float, complex, str, bytes]], Sequence[Sequence[Union[bool, int, float, complex, str, bytes]]], Sequence[Sequence[Sequence[Union[bool, int, float, complex, str, bytes]]]], Sequence[Sequence[Sequence[Sequence[Union[bool, int, float, complex, str, bytes]]]]]]]], side: Union[Literal['left'], Literal['right']] = ..., sorter: Union[_SupportsArray[dtype[Union[bool_, integer[Any]]]], Sequence[_SupportsArray[dtype[Union[bool_, integer[Any]]]]], Sequence[Sequence[_SupportsArray[dtype[Union[bool_, integer[Any]]]]]], Sequence[Sequence[Sequence[_SupportsArray[dtype[Union[bool_, integer[Any]]]]]]], Sequence[Sequence[Sequence[Sequence[_SupportsArray[dtype[Union[bool_, integer[Any]]]]]]]], int, Sequence[int], Sequence[Sequence[int]], Sequence[Sequence[Sequence[int]]], Sequence[Sequence[Sequence[Sequence[int]]]], None] = ...) -> ndarray[Any, dtype[signedinteger[Any]]]
pandas/core/groupby/groupby.py:2949: error: Incompatible types in assignment (expression has type "Union[ExtensionArray, ndarray[Any, Any]]", variable has type "ndarray[Any, Any]")  [assignment]
pandas/core/groupby/groupby.py:2984: error: Incompatible types in assignment (expression has type "Union[ExtensionArray, ndarray[Any, Any]]", target has type "ndarray[Any, Any]")  [assignment]
```

--- a/pandas/core/groupby/groupby.py
+++ b/pandas/core/groupby/groupby.py
@@ -2896,7 +2896,7 @@ class GroupBy(BaseGroupBy[FrameOrSeries]):
         grouper = self.grouper
 
         ids, _, ngroups = grouper.group_info
-        output: dict[base.OutputKey, np.ndarray] = {}
+        output: dict[base.OutputKey, ArrayLike] = {}
 
         base_func = getattr(libgroupby, how)
         base_func = partial(base_func, labels=ids)
@@ -2911,6 +2911,7 @@ class GroupBy(BaseGroupBy[FrameOrSeries]):
             else:
                 result_sz = len(values)
 
+            result: ArrayLike
             result = np.zeros(result_sz, dtype=cython_dtype)
             if needs_2d:
                 result = result.reshape((-1, 1))

--- a/pandas/core/reshape/tile.py
+++ b/pandas/core/reshape/tile.py
@@ -418,7 +418,11 @@ def _bins_to_cuts(
             bins = unique_bins
 
     side = "left" if right else "right"
-    ids = ensure_platform_int(bins.searchsorted(x, side=side))
+    # error: No overload variant of "searchsorted" of "ndarray" matches
+    # argument types "Any", "str"
+    ids = ensure_platform_int(
+        bins.searchsorted(x, side=side)  # type: ignore[call-overload]
+    )
 
     if include_lowest:
         ids[np.asarray(x) == bins[0]] = 1

